tests/iostability/test_iostability_degraded_path.py

```diff
@@ -116,3 +121,56 @@ def test_object_crud_single_pod_failure(self):
         self.test_completed = True
         self.log.info("ENDED: Test for Object CRUD operations in degraded mode in loop using "
                       "S3bench for 7 days")
+
+    @pytest.mark.lc
+    @pytest.mark.io_stability
+    @pytest.mark.tags("TEST-40173")
+    def test_disk_near_full_read_in_degraded_s3bench(self):
+        """
+        Perform disk storage near full once in healthy cluster and read in degraded cluster in loop
+        for 7 days.
+        """
+        self.log.info("STARTED: Perform disk storage near full once in healthy cluster and "
+                      "read in degraded cluster in loop for 7 days.")
+        s3userinfo = dict()
+        s3userinfo['accesskey'] = ACCESS_KEY
+        s3userinfo['secretkey'] = SECRET_KEY
+        bucket_prefix = "testbkt"
+        duration_in_days = self.test_cfg['degraded_path_durations_days']
+        client = len(self.worker_node_list) * self.test_cfg['sessions_per_node_vm']
+        percentage = self.test_cfg['test_40173']['nearfull_memory_percetange']
+        self.log.info("Step 1: calculating byte count for required percentage")
+        resp = self.dfr.get_user_data_space_in_bytes(master_obj=self.master_node_list[0],
+                                                     memory_percent=percentage)
+        assert_utils.assert_true(resp[0], resp[1])
+        self.log.info("Need to add %s bytes for required percentage", resp[1])
+        self.log.info("Step 2: performing writes till we reach required percentage")
+
+        ret = DiskFailureRecoveryLib.perform_near_full_sys_writes(s3userinfo=s3userinfo,
+                                                                  user_data_writes=int(resp[1]),
+                                                                  bucket_prefix=bucket_prefix,
+                                                                  client=client)
+        assert_utils.assert_true(ret[0], ret[1])
+        self.log.debug("write operation data: %s", ret)
+        self.log.info("Step 2: Shutdown the data pod safely by making replicas=0, "
+                      "check degraded status.")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(self.master_node_list[0],
+                                                             self.hlth_master_list[0])
```

Comment:
No need to recover deleted pod? Or you are going to redeploy after each test case?

Reply:
Yes, we will be redeploying setup manually.

Reply:
I think we should continue on the same deployment. Recovering the pods is feasible as you are not tampering data at rest. Also the data is already written in original SNS config. Note that if you would write on degraded mode you would read data written in degraded mode as well.

Reply:
write scope of F-99D/E is only single pod shutdown and as these are long-running test case we are not planning to redeploy setup automatically, as we might loose important data which might not be part of support bundle.

Reply:
Well pre fill data once created in near full suite level setup can be reused till the failure occurs. This can save many hours of execution.